03week/towersOfHanoi.js

```diff
@@ -17,28 +23,85 @@ function printStacks() {
   console.log("a: " + stacks.a);
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
+  // console.log("and so it begins....");
+  // console.log(startStack[startStack.length-1]);
 }
 
-function movePiece() {
-  // Your code here
 
+// ========================================================
+// ========= Checkpoint1 Assignment Starts Here ===========
+// ========================================================
+
+
+// ========== Function movePiece =========
+function movePiece(startStack, endStack) {
+  // move the top piece from startStack to top of endStack
+  let movingPiece = stacks[startStack].pop();
+  stacks[endStack].push(movingPiece);
+  // console.log("moved piece and updated the board");
 }
 
-function isLegal() {
-  // Your code here
 
+// ========= Function isLegal =========
+function isLegal(startStack, endStack) {
+  // if the move is legal return true, a legal move is
```

Comment:
what if they type in an invalid name for the stack

Reply:
noting would happen, the code was only set up for exact match, it actually throws an error though.